packages/page-storage/src/Selection/Raw.tsx

```diff
@@ -27,30 +36,94 @@ function Raw ({ onAdd }: Props): React.ReactElement<Props> {
 
       setValue({
         isValid: u8a.length !== 0,
-        key: compactAddLength(u8a)
+        key,
+        keyAsBytes: compactAddLength(u8a)
       });
     },
     []
   );
 
+  const allStoragePrefixes = useMemo(
+    () => {
+      const palletNames = Object
+        .keys(api.query)
+        .map((value): string => (
+          value
+        ));
+      const allStoragePrefixesNotFlattened = palletNames.map((palletName) => {
+        const pallet = api.query[palletName];
+
+        return Object
+          .keys(pallet)
+          .map((palletCall) => {
+            const { keyPrefix } = pallet[palletCall] as unknown as StorageEntry;
+            const keyPrefixValue = keyPrefix().toString();
+
+            return [keyPrefixValue, `${palletName}.${palletCall}`];
+          });
+      });
+
+      return allStoragePrefixesNotFlattened.flat();
+    },
+    [api]
+  );
+
+  const decodedKey = useMemo(
+    (): string => {
+      if (isValid) {
+        const matchedPrefixes = allStoragePrefixes.filter(([keyPrefix, _]) => {
+          return keyPrefix.includes(key);
```

Comment:
dont you want to use `startsWith` here?

Reply:
user can type `0x...` which will work like `startsWith`, and now we have more flexibility.